python_modules/libraries/dagster-aws/dagster_aws/ecs/tasks.py

```diff
@@ -33,27 +135,49 @@ class EcsNoTasksFound(Exception):
     pass
 
 
-def default_ecs_task_definition(
+def get_task_definition_dict_from_current_task(
     ecs,
     family,
-    metadata,
+    current_task,
     image,
     container_name,
     environment,
+    execution_role_arn,
+    task_role_arn,
+    log_group,
+    sidecar_container_definitions,
     command=None,
     secrets=None,
     include_sidecars=False,
 ):
+
+    current_container_name = current_ecs_container_name()
+
+    current_task_definition_arn = current_task["taskDefinitionArn"]
+    current_task_definition_dict = ecs.describe_task_definition(
+        taskDefinition=current_task_definition_arn
+    )["taskDefinition"]
+
+    container_definition = next(
+        iter(
+            [
+                container
+                for container in current_task_definition_dict["containerDefinitions"]
+                if container["name"] == current_container_name
+            ]
+        )
+    )
+
     # Start with the current process's task's definition but remove
     # extra keys that aren't useful for creating a new task definition
     # (status, revision, etc.)
     expected_keys = [
         key for key in ecs.meta.service_model.shape_for("RegisterTaskDefinitionRequest").members
     ]
     task_definition = dict(
-        (key, metadata.task_definition[key])
+        (key, current_task_definition_dict[key])
         for key in expected_keys
-        if key in metadata.task_definition.keys()
+        if key in current_task_definition_dict.keys()
```

Comment:
@jmsanders my understanding is that the answer to your question of what else can end up in the task definition if you set use_current_task_definition=True can be found here - it's the set of things that are a) params going into register_task_definition and b) found on the output of describe_task_definition

Reply:
But then we merge a bunch of stuff onto it based on what's in your launcher config.

I guess my question is how much is even left that we actually inherit vs. override.

Reply:
There are many random fields here on the taskDefinition arg to  that would no longer be copied over (placementConstraints etc.): https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecs.html#ECS.Client.register_task_definition

Whether they are actually in use by anybody, I do not know